### ISSUE TITLE
Nexo Print Server Local printing - Default printer not applicable when you have two or more printers 

### ISSUE THREAD

**Issue by mbugikp** [Blair2004/NexoPOS#1478]
**Describe the bug**
The setup is bar, restaurant, and kitchen each has its own printer so we have three printers using NPS for local printing.

**To Reproduce**
Have a kitchen printer, add a bar printer, add a restaurant printer, the kitchen printer will printer kitchen orders, the bar printer will print bar orders and the restaurant printer will print restaurant orders. NPS requires a default printer choose any printer for my case I will choose restaurant printer as default printer.

Since I am using Gastro I will send an order to the Kitchen and it gets printed correctly, I go to the bar on first floor make an order it gets printed correctly on the bar printer, I proceed to print the bill/receipt it gets printed down stairs by the restaurant printer which happens to be the default printer. Now the waiter has to go down stairs to get the customer bill/receipt from the restaurant printer wasting a lot of time.

**Expected behavior**
Waiter should be able to print customer bill/receipt from bar printer, NPS should only print through the default printer if waiter did not choose a printer, therefore we should provide waiter with a list of available printers to choose from before printing is completed. You can imagine leaving a printer on the first floor (bar) to go and pick a bill/receipt on the ground floor(restaurant).

 
**Screenshots**

**Environment:**
 - NexoPOS 4.8.17
 - Premium Extensions: [e.g Nexo Print Adapter (4.8.6), Gastro (4.8.15)

**Comment by ahorasoyfelipe** [comment]
hello, we have solved that using the configuration of each computer example: 

1 computer in the cashier has 3 connections to the printers, one is called the bar, another kitchen and another cashier

In the 2 computers bar and kitchen are configured in the same way 

Cashier : 192.168.123.150
 bar : 192.168.123.160
kitchen: 192.168.123.170

but on a second pc the configuration of that pc is:

cashier: connected by usb

therefore if I print from pc 1, it appears on the cashier printer that is on the same pc and on the second pc, the system reads that the printer is still called a cashier, but it prints it on the printer installed on that pc

In short, use the same printer names on each PC, but on each PC when you install the printers you can configure them with different ports

**Comment by mbugikp** [comment]
Thank you @ahorasoyfelipe but I am getting lost.
1) Do you have nexo print server? If yes are you using local printing?
2) What is your default printing gateway? Web or nps?
![POS Settings printing gateway](https://github.com/Blair2004/NexoPOS/assets/13999977/17136726-92f8-447b-ba6f-3becaaf2686e)



3) I am using network printers for restaurant and kitchen, the bar printer is connected by usb, will this scenario still work?

**Comment by Blair2004** [comment]
>  I proceed to print the bill/receipt it gets printed down stairs by the restaurant printer which happens to be the default printer.

Where do you "proceed to print"? This will help us understand which section still use the default printer.

**Comment by mbugikp** [comment]
proceed to print
![Print_order](https://github.com/Blair2004/NexoPOS/assets/13999977/9a54a0b6-a2be-4844-b903-63416209b488)

**Comment by Blair2004** [comment]
What version of NPS Adapter your using ? It's not mentionned

**Comment by mbugikp** [comment]
NPS Adapter 4.8.6

**Comment by Blair2004** [comment]
Use 4.8.7 as that version nows allows you to select the printer before printing.

**Comment by mbugikp** [comment]
Okay. Let me check.

**Comment by Blair2004** [comment]
Comment if the issue persists
